src/wp-includes/functions.php

```diff
@@ -647,13 +647,34 @@ function maybe_serialize( $data ) {
  * Unserializes data only if it was serialized.
  *
  * @since 2.0.0
+ * @since 6.8.0 Added checking for unserialize warnings including PHP 8.3's trailing data warnings.
```

Comment:
This description feels a bit off. In intent, nothing has changed through the use of the error handler. What was happening before is that we used the error-suppression operator to hide the warnings, effectively ignoring them as we still do, but that unfortunately left them doing weird things, showing up in logs, etc…

This change improves our implementation of that goal: hide and ignore any issues during unserialization. To that end, I don’t know if we need this message, and if we do, maybe we note instead that this now returns `false` when serialized classes crash during wake-up. Previously they would crash the request.